enterprise/web.jsf/src/org/netbeans/modules/web/jsf/wizards/PersistenceClientIterator.java

```diff
@@ -632,8 +632,18 @@ private static boolean showImportStatement(String packageName, String fqn) {
     }
 
     private static boolean isCdiEnabled(Project project) {
-        CdiUtil cdiUtil = project.getLookup().lookup(CdiUtil.class);
-        return (cdiUtil == null) ? false : cdiUtil.isCdiEnabled();
+        WebModule wm = WebModule.getWebModule(project.getProjectDirectory());
```

Comment:
Can really be both `CdiUtils` in the lookup? If not, I would ditch the check here and just try to lookup both sequentially and the first that is found, is used.

Reply:
I believe we would receive a class not found exception if using Jakarta EE 9+ and returning a org.netbeans.modules.web.beans.CdiUtil.class, so I think it is a good idea to perform the check to see which version of Jakarta EE/Java EE is in use by the project.

Reply:
I don't see an issue, this should work:

```java
    private static boolean isCdiEnabled(Project project) {
        org.netbeans.modules.jakarta.web.beans.CdiUtil jakartaCdiUtil = project.getLookup().lookup(org.netbeans.modules.jakarta.web.beans.CdiUtil.class);
        if(jakartaCdiUtil != null && jakartaCdiUtil.isCdiEnabled()) {
            return true;
        }
        org.netbeans.modules.web.beans.CdiUtil javaxCdiUtil = project.getLookup().lookup(org.netbeans.modules.web.beans.CdiUtil.class);
        if(javaxCdiUtil != null && javaxCdiUtil.isCdiEnabled()) {
            return true;
        }
        return false;
    }
```

If the classes could not be loaded, it would blow up anyway as `PersistenceClientIterator` could not be loaded then. For exceptions from the `lookup` method:

https://bits.netbeans.org/17/javadoc/org-openide-util-lookup/org/openide/util/Lookup.html#lookup-java.lang.Class-

> **Returns:**
> an object implementing the given class or null if no such implementation is found

So if the corresponding CdiUtil is present, then it is used else it is null and not checked.

Reply:
I see what you are explaining now and that will work well.  I appreciate the feedback.